### PR TITLE
Fix flyout interaction & re-enable multi-flyout caching

### PR DESCRIPTION
**Pull request by darzu**
Reverts & fixes: https://github.com/Microsoft/pxt/pull/5337

My multi-flyout caching introduced with #5323 broke dragging of blocks from the flyout after it was closed then opened.

This was due to destructive side-effects in the Flyout.hide() code in blockly, which I've proposed a fix for here: https://github.com/Microsoft/pxt-blockly/pull/212

Try the fix here: https://arcade.makecode.com/app/e07373f9b0f4944f2a118372b04b2ae7550353b3-0f3b278df6#editor

Validation:
- [x] Ensure leaving event registrations on hide() doesn't cause further issues
  - confirmed the only events that are registered are: `mousedown, mouseenter, mouseleave, mousedown` which don't trigger on display: none
- [x] complete a tutorial on IE 11

### CODE REVIEW

**Comment by darzu** [comment]
Found an issue with zooming in & out. Only the active flyout gets zoomed.
- [ ] Fix zoom in & out issue

File: webapp/src/blocks.tsx [hunk 1]
```diff
@@ -1287,11 +1288,43 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         this.showFlyoutInternal_(this.flyoutXmlList);
     }
 
-    private showFlyoutInternal_(xmlList: Element[]) {
+    private flyouts: pxt.Map<Blockly.VerticalFlyout> = {};
```
Comment (pelikhan) [inline webapp/src/blocks.tsx]: When do we clear this cache?

Comment (darzu) [reply webapp/src/blocks.tsx]: So far, never. I can't think of a case where we'd want to. This cache doesn't grow unbounded, there's only one one entry per toolbox category. We refresh the contents of the flyouts in these caches when appropriate. Some flyouts that change a lot like variables we don't even cache.

Comment (darzu) [reply webapp/src/blocks.tsx]: It's not exactly a cache. We're just maintaining multiple flyouts at once, and they're mutable.

**Comment by pelikhan** [comment]
You need to clear the cache whenever a project gets reload as the fly out might change , e.g. an extension adds new blocks.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pelikhan** [comment]
I think there is already a place where we blow out caches

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pelikhan** [comment]
Try adding an extension, open fly out, remove extension, open fly out.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by darzu** [comment]
> Try adding an extension, open fly out, remove extension, open fly out. Get Outlook for iOS<https://aka.ms/o0ukef>

I see. I forgot extensions could add blocks to existing flyouts

**Comment by pelikhan** [comment]
Also after creating a new variable, blocks might change to autoincrment the block name.

**Comment by darzu** [comment]
> Also after creating a new variable, blocks might change to autoincrment the block name.

I can't find an example of this outside the Variables flyout (which works). Do you know of any?

**Comment by darzu** [comment]
> > Also after creating a new variable, blocks might change to autoincrment the block name.
> 
> I can't find an example of this outside the Variables flyout (which works). Do you know of any?

Ah found it. Creating the variable "mySprite" causes the sprite blocks to increment to "mySprite2". This is subtle stuff.

**Comment by darzu** [comment]
I've implemented a finer-grain cache invalidation scheme based on the hash of the blocks xml. This seems to work very well and handles all cases I know of.
Test here:
https://arcade.makecode.com/app/a5970054f60bc4ab93403bc282041065a6ad70db-205a1c8059

**Comment by darzu** [comment]
- [ ] pull in the latest pxt-blockly changes

**Comment by darzu** [comment]
Found an issue with the caching only in IE.
- [ ] fix IE issue

**Comment by darzu** [comment]
> I think you should still clean the cached when a header is reloaded.

Why is that? Since this is caching based on the hash of the blocks xml, it should be plenty precise and will save extra work. The first load can be really slow, on the order of several seconds, up to 10 seconds. Avoid re-doing that work as much as possible is very good thing in my mind. I don't see a case where invalidating on header reload would catch anything that invalidating on the xml hash wouldn't.

**Comment by darzu** [comment]
> Found an issue with the caching only in IE. Works in Firefox, Safari, and Chrome..
> 
> * [x]  fix IE issue

Fixed. Here: https://arcade.makecode.com/app/fbaae575ebff2f139efa7b5b80c0a4c040cb371c-4f44116f61#editor

**Comment by darzu** [comment]
Test here: https://arcade.makecode.com/app/6b23f6639d69a49656d0af6dc6495a1b82019220-0bbe4f1578

**Pull request merged**